epdb/epdb_client.py

```diff
@@ -143,13 +146,20 @@ def interact(self):
                         print('*** Connection closed by remote host ***')
                         break
                     if text:
-                        sys.stdout.write(text.decode('ascii'))
+                        all_text = remaining_text + text
+                        try:
+                            unicode_text = all_text.decode(encoding)
+                            remaining_text = b""
+                        except UnicodeDecodeError as e:
+                            unicode_text = all_text[:e.start].decode(encoding)
```

Comment:
Would it make sense to check that `e.start > 0`? Otherwise you may end up in an infinite loop?

Reply:
Wouldn't that happen only if buffer doesn't fit a single Unicode character that we get it of 0? (Very unlikely) and I don't think it could be negative. Overall- I don't think there is a realistic way into getting into infinite loop. I could be proven wrong

Reply:
```
>>> len(b"\600")
1
>>> print(b"\600".decode("utf-8"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte
```

In other words, if the very first byte happens to be an invalid UTF-8 starting multi-char rune, it will return zero. Yes, it's very much a corner case, but I think the extra `if` is cheap assurance.

Reply:
sure, can add.. but still curious if needed.  So what if the `e.start` is 0? code above should be just as fine AFAIK.  Then the question is either we should guard for `<0` but then if so -- do what?  if to error -- then smth like  
```suggestion
                            assert e.start < 0, f"Got an unexpected negative start for unicode: {e.start}"
                            unicode_text = all_text[:e.start].decode(encoding)
```
?